### PR TITLE
*: fix formatting errors with llvm

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This fixes a bunch of formatting errors picked up by compiling with `llvm`

### Issues/PRs references
Detected in #9398.

### CODE REVIEW

**Comment by miri64** [comment]
It appears our ARM7 platforms don't like the `PRIu32` macro :-/... So I used the casting method instead.

**Comment by gebart** [comment]
Why would the arm7 be different from the cortexes? Are they not using newlib too?

**Comment by miri64** [comment]
> Why would the arm7 be different from the cortexes? Are they not using newlib too?

I believe the main difference is, that they don't use newlib-nano (?). The symptom I saw, was that while `PRIu32` was defined it seemed to be empty: https://ci.riot-os.org/RIOT-OS/RIOT/9677/6f5ebb3a84402b93a74c80ede54fa4c4ef70d18f/output/compile/tests/driver_sdcard_spi/avsextrem.txt

**Comment by miri64** [comment]
> I see, does #6671 do anything for this issue?

No. Neither does it fix the errors I get for ARM7 boards in earlier versions of this PR (and why would it, the ARM7 boards aren't build with LLVM (yet)?) nor does it fix the formatting errors observed in #9398.

**Comment by miri64** [comment]
Mh... including the header that defines these macros also helps ;-). I reverted my fix-ups and included instead.

**Comment by gebart** [comment]
Some of the files are still missing include inttypes.h

**Comment by miri64** [comment]
> Some of the files are still missing include inttypes.h

Fixed

File: drivers/mtd_sdcard/mtd_sdcard.c [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #include "sdcard_spi.h"
 #include "sdcard_spi_internal.h"
 
-#include <stdint.h>
+#include <inttypes.h>
```
Comment (miri64) [inline drivers/mtd_sdcard/mtd_sdcard.c]: `inttypes.h` [includes `<stdint.h>`](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/inttypes.h.html#tag_13_20) that's why I removed it.

**Comment by miri64** [comment]
Ping @gebart?

File: tests/driver_io1_xplained/main.c [hunk 2]
```diff
@@ -43,10 +43,10 @@ static void _sd_card_cid(void)
...
            dev.sdcard.cid.PNM[0], dev.sdcard.cid.PNM[1], dev.sdcard.cid.PNM[2],
            dev.sdcard.cid.PNM[3], dev.sdcard.cid.PNM[4]);
-    printf("PRV: %d\n", dev.sdcard.cid.PRV);
-    printf("PSN: %lu\n", dev.sdcard.cid.PSN);
-    printf("MDT: %d\n", dev.sdcard.cid.MDT);
-    printf("CRC: %d\n", dev.sdcard.cid.CID_CRC);
+    printf("PRV: %u\n", dev.sdcard.cid.PRV);
+    printf("PSN: %" PRIu32 "\n", dev.sdcard.cid.PSN);
+    printf("MDT: %u\n", dev.sdcard.cid.MDT);
+    printf("CRC: %u\n", dev.sdcard.cid.CID_CRC);
```
Comment (cladmi) [inline tests/driver_io1_xplained/main.c]: `PRV`, `MDT` and `CID_CRC` do not raise any warning but they are indeed `unsigned` types in `drivers/include/sdcard_spi.h`.

Comment (miri64) [reply tests/driver_io1_xplained/main.c]: That's why (though unrelated) I changed them ;-).

**Comment by miri64** [comment]
Squashed

File: tests/driver_io1_xplained/main.c (hunk 2 above)
Comment (cladmi) [reply tests/driver_io1_xplained/main.c]: Agreed

**Pull request merged**